### ISSUE TITLE
Handling location.hash

### ISSUE THREAD

**Issue by RUSSCITY** [krasimir/navigo#215]
My router looks like this: 
```
this.router.on({
  "/faq": () => {
     viewChannel.request("showPage", PAGES.faq);
   },
   "*": () => {
     viewChannel.request("showPage", PAGES.not_found);
   }
}).resolve();
```

This works fine on `mysite.com/faq`, but as soon as I add some hash to url, I cannot trigger this route anymore. Like `mysite.com/faq#anchor` fail in "not found" section. 
I tried to fix it with some regular expressions, but with no success. Can anyone help me?

**Comment by krasimir** [comment]
Hello, @RUSSCITY how did you initialize the router?

**Comment by RUSSCITY** [comment]
Hello, @krasimir, thank you for writing back!

I initialize router without any parameters: 
```
import Navigo from "navigo";

export class MainRouter {
  constructor() {
    this.router = new Navigo();
    this.initializeRoutes();
  }
  initializeRoutes() {
    this.router
      .on({
        "/faq": () => {
          viewChannel.request("showPage", PAGES.faq);
        },
        "*": () => {
          viewChannel.request("showPage", PAGES.not_found);
        }
   }).resolve();
  }
}
```

My backend is configured to always show the same frontend ignoring what is in url, like `mysite.com/faq` and `mysite.com/xyz` or `mysite.com/faq#somewhatelse` would always open the page with the router above. You can see the problem "live" here: https://alltracker.org/faq vs https://alltracker.org/faq#

**Comment by lnorton89** [comment]
I'm currently working on a workaround for this issue but am stuck. Here is my solution:

`router
  .on('pagename/:section', function (params) {
    functions.loadHTML('./dist/templates/pagename.html', 'view');
    //Use params.section programatically to scroll to that position on page. 
  })
  .resolve();`

This solves the issue but I can not access the page by just localhost/#!pagename anymore, only with params (localhost/#1pagename/sectionname). I guess I could just add another route with pagename but this seems silly.

**Comment by krasimir** [comment]
This issue is a duplicate of this one https://github.com/krasimir/navigo/issues/111#issuecomment-750402327 Which is fixed in the new 8.0.0 version.

P.S.
Version 8.0.0 is not officially released but can be installed via `npm install navigo@beta` or `yarn add navigo@beta`. Migration guide is available here https://github.com/krasimir/navigo/blob/big-rewrite/CHANGELOG.md#migration-guide and the new documentation here https://github.com/krasimir/navigo/blob/big-rewrite/DOCUMENTATION.md.
